File: _정답코드/5주차/09_누적_점수_구간_XOR_게임/solution.py
def solution(numbers, queries):
    n = len(numbers)
    prefixXor = [0] * (n + 1)
    
    for i in range(n):
        prefixXor[i + 1] = prefixXor[i] ^ numbers[i]
    
    result = []
    for i in range(len(queries)):
        l = queries[i][0]
        r = queries[i][1]
        result.append(prefixXor[r] ^ prefixXor[l - 1])
    
    return result

File: _정답코드/5주차/09_누적_점수_구간_XOR_게임/Solution.java
class Solution {
    public int[] solution(int[] numbers, int[][] queries) {
        int n = numbers.length;
        int[] prefixXor = new int[n + 1];
        
        for (int i = 0; i < n; i++) {
            prefixXor[i + 1] = prefixXor[i] ^ numbers[i];
        }
        
        int[] result = new int[queries.length];
        for (int i = 0; i < queries.length; i++) {
            int l = queries[i][0];
            int r = queries[i][1];
            result[i] = prefixXor[r] ^ prefixXor[l - 1];
        }
        
        return result;
    }
}

